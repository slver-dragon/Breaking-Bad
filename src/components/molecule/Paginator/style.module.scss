@import "../../../assets/styles/style";
.main {
  display: flex;
  align-items: center;
  @include paginatorMain;
  .pages {
    display: flex;
  }
  .elementPerPage {
    display: flex;
    @include mediumFont;
    align-items: center;
  }
  span {
    margin: 0 auto;
    padding: 0.29rem;
  }
  .active {
    color: $yellow;
    @include shadowBox;
    padding: 0.29rem;
  }
}
