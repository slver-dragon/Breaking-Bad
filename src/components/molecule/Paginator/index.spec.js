import React from "react";
import { shallow } from "enzyme";
import Paginator from ".";

describe("Testing molecule Paginator:", () => {
  let component = null;
  const onBtnClick = jest.fn();
  const setComponent = (props) => shallow(<Paginator {...props} />);

  test("is rendering", () => {
    component = setComponent({
      pageElementCount: 10,
      cardsMaxCount: 63,
      currentPage: 2,
    });
    expect(component).toMatchSnapshot();
  });

  test("change page to left", () => {
    component = setComponent({ changePage: onBtnClick, currentPage: 2 });
    component.find(".pages").childAt(0).simulate("click");
    expect(onBtnClick).toHaveBeenCalled();
  });

  test("change page to right", () => {
    component = setComponent({ changePage: onBtnClick, currentPage: 2 });
    component.find(".pages").childAt(1).simulate("click");
    expect(onBtnClick).toHaveBeenCalled();
  });

  test("count pages calculate", () => {
    component = setComponent({
      pageElementCount: 10,
      cardsMaxCount: 63,
      currentPage: 2,
    });
    expect(component.find(".pages").children().length).toEqual(9);
  });

  test("pages list click", () => {
    component = setComponent({
      changePage: onBtnClick,
      pageElementCount: 5,
      cardsMaxCount: 63,
      currentPage: 6,
    });
    component.find(".active").at(0).simulate("click");
    expect(onBtnClick).toHaveBeenCalled();
    component.find("#CurrentPage").at(0).simulate("click");
    expect(onBtnClick).toHaveBeenCalled();
    component.find("#StartPage").at(0).simulate("click");
    expect(onBtnClick).toHaveBeenCalled();
    component.find("#EndPage").at(0).simulate("click");
    expect(onBtnClick).toHaveBeenCalled();
    component.find("#MultiDotFirst").simulate("click");
    expect(onBtnClick).toHaveBeenCalled();
    component.find("#MultiDotLast").simulate("click");
    expect(onBtnClick).toHaveBeenCalled();
  });

  test("NumberElement list click", () => {
    component = setComponent({ changeCount: onBtnClick, pageElementCount: 10 });
    component.find(".active").simulate("click");
    expect(onBtnClick).toHaveBeenCalled();
    component.find("#CountElement").at(0).simulate("click");
    expect(onBtnClick).toHaveBeenCalled();
  });

  test("EndPage absent", () => {
    component = setComponent({
      changePage: onBtnClick,
      pageElementCount: 10,
      cardsMaxCount: 63,
      currentPage: 5,
    });
    expect(component.find("#EndPage").length).toEqual(0);
  });
});
