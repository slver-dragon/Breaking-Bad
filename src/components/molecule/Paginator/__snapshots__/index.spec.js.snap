// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing molecule Paginator: is rendering 1`] = `
<div
  className="main"
>
  <div
    className="pages"
  >
    <span
      id="CurrentPage"
      key="1"
      onClick={[Function]}
    >
      <Button
        text={1}
      />
    </span>
    <span
      className="active"
      id="CurrentPage"
      key="2"
      onClick={[Function]}
    >
      <Button
        text={2}
      />
    </span>
    <span
      id="CurrentPage"
      key="3"
      onClick={[Function]}
    >
      <Button
        text={3}
      />
    </span>
    <span
      id="CurrentPage"
      key="4"
      onClick={[Function]}
    >
      <Button
        text={4}
      />
    </span>
    <span
      id="MultiDotLast"
      onClick={[Function]}
    >
      <Button
        text="..."
      />
    </span>
    <span
      id="EndPage"
      key="6"
      onClick={[Function]}
    >
      <Button
        text={6}
      />
    </span>
    <span
      id="EndPage"
      key="7"
      onClick={[Function]}
    >
      <Button
        text={7}
      />
    </span>
    <span
      onClick={[Function]}
    >
      <Button
        img="test-file-stub"
      />
    </span>
    <span
      onClick={[Function]}
    >
      <Button
        img="test-file-stub"
      />
    </span>
  </div>
  <div
    className="elementPerPage"
  >
    <p>
      Отобразить карточек:
    </p>
    <span
      id="CountElement"
      key="5"
      onClick={[Function]}
    >
      <Button
        text={5}
      />
    </span>
    <span
      className="active"
      id="CountElement"
      key="10"
      onClick={[Function]}
    >
      <Button
        text={10}
      />
    </span>
    <span
      id="CountElement"
      key="15"
      onClick={[Function]}
    >
      <Button
        text={15}
      />
    </span>
    <span
      id="CountElement"
      key="20"
      onClick={[Function]}
    >
      <Button
        text={20}
      />
    </span>
  </div>
</div>
`;
